### PR TITLE
fscrypt-tests.py: adding test

### PR DESCRIPTION
**Pull request by nasastry**
Adding fscrypt tests, fscrypt is a tool for managing Linux file system
encryption

Signed-off-by: Nageswara R Sastry <rnsastry@linux.ibm.com>

### CODE REVIEW

**Comment by nasastry** [comment]
```
Fetching asset from fscrypt-tests.py:fscrypt.test
JOB ID     : 4e6c3c9cc14cfddafd107f83ed5eb2fadda33431
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-06T03.25-4e6c3c9/job.log
 (1/1) fscrypt-tests.py:fscrypt.test: CANCEL: CONFIG_FS_ENCRYPTION not set. (0.52 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-06T03.25-4e6c3c9/results.html
JOB TIME   : 1.71 s

Fetching asset from fscrypt-tests.py:fscrypt.test
JOB ID     : 27013503903f7be9e240c171b75fa9af001b0268
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2021-07-06T03.34-2701350/job.log
 (1/1) fscrypt-tests.py:fscrypt.test: PASS (7.39 s)
RESULTS    : PASS 1 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 0
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2021-07-06T03.34-2701350/results.html
JOB TIME   : 7.85 s
```

File: security/fscrypt-tests.py [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        if detected_distro.name in ['rhel', 'SuSE', 'fedora', 'centos']:
+            deps.extend(["autoconf", "automake", "golang", "expect",
+                         "keyutils", "e2fsprogs", "pam-devel"])
+        else:
+            self.cancel("Unsupported distro %s for fscrypt package"
+                        % detected_distro.name)
+        for package in deps:
+            if not smm.check_installed(package) and not smm.install(package):
+                self.cancel('%s is needed for the test to be run' % package)
+        url = "https://github.com/google/fscrypt/archive/refs/heads/master.zip"
```
Comment (PraveenPenguin) [inline security/fscrypt-tests.py:53]: ah it tricky , as above source has Apache-2.0 License  and in misc test repo we have GPU  License  not sure we can call this from this repo @narasimhan-v can you please suggest , 

@nasastry did not started review , let us first have more clarity on  above query than we can have review and merge

Comment (Naresh-ibm) [reply security/fscrypt-tests.py:53]: @PraveenPenguin @nasastry any further discussion on this?

Comment (abdhaleegit) [reply security/fscrypt-tests.py:53]: @PraveenPenguin we are fine with GPL and Apache License, both are almost similar open source free software liscence

**Comment by nasastry** [comment]
This test case applies only to `upstream` and not to `distro`.

```
RHEL:
JOB ID     : 906ea5980cdccc757285ae9e3ca162d9ba163d87
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-19T03.56-906ea59/job.log
 (1/1) fscrypt-tests.py:fscrypt.test: CANCEL: CONFIG_FS_ENCRYPTION not set. (0.47 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-19T03.56-906ea59/results.html
JOB TIME   : 13.48 s

SLES:
JOB ID     : 478a342557a81a76eff77d9ea20e051327aeb151
JOB LOG    : /home/avocado-fvt-wrapper/results/job-2022-10-19T14.23-478a342/job.log
 (1/1) fscrypt-tests.py:fscrypt.test: CANCEL: golang is needed for the test to be run (0.91 s)
RESULTS    : PASS 0 | ERROR 0 | FAIL 0 | SKIP 0 | WARN 0 | INTERRUPT 0 | CANCEL 1
JOB HTML   : /home/avocado-fvt-wrapper/results/job-2022-10-19T14.23-478a342/results.html
JOB TIME   : 22.65 s
```

[rhel_fscrypt_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9818797/rhel_fscrypt_job.log)
[sles_fscrypt_job.log](https://github.com/avocado-framework-tests/avocado-misc-tests/files/9818800/sles_fscrypt_job.log)

File: security/fscrypt-tests.py [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+#
+# See LICENSE for more details.
+#
+# Copyright: 2021 IBM
```
Comment (abdhaleegit) [inline security/fscrypt-tests.py:14]: 2022 ?

**Review by abdhaleegit — APPROVED** [review]

**Pull request merged**